src/main/java/org/sagebionetworks/bridge/services/OrganizationService.java

```diff
@@ -213,50 +211,45 @@ public void deleteOrganization(String appId, String identifier) {
             .withAdminOnly(true)
             .withOrgMembership("<none>").build();
 
-        return accountDao.getPagedAccountSummaries(appId, scopedSearch);
+        return accountService.getPagedAccountSummaries(appId, scopedSearch);
     }
     
     /**
      * Once assigned, only admins can re-assign accounts.
      */
-    public void addMember(String appId, String identifier, AccountId accountId) {
+    public void addMember(String appId, String identifier, String userId) {
         checkArgument(isNotBlank(appId));
         checkArgument(isNotBlank(identifier));
-        checkNotNull(accountId);
+        checkArgument(isNotBlank(userId));
         
         CAN_EDIT_MEMBERS.checkAndThrow(ORG_ID, identifier);
         
-        Account account = accountDao.getAccount(accountId)
-                .orElseThrow(() -> new EntityNotFoundException(Account.class));
-        
-        RequestContext context = RequestContext.get();
-        if (!context.isInRole(ADMIN) && account.getOrgMembership() != null) {
-            throw new BadRequestException("Account already assigned to an organization.");
-        }
-        
-        account.setOrgMembership(identifier);
-        accountDao.updateAccount(account);
-        sessionUpdateService.updateOrgMembership(account.getId(), identifier);
+        AccountId accountId = AccountId.forId(appId, userId);
+        accountService.editAccount(accountId, (acct) -> {
+            RequestContext context = RequestContext.get();
+            if (!context.isInRole(ADMIN) && acct.getOrgMembership() != null) {
+                throw new BadRequestException("Account already assigned to an organization.");
+            }
+            acct.setOrgMembership(identifier);
+        });
+        sessionUpdateService.updateOrgMembership(userId, identifier);
     }
```

Comment:
The use of userId in this last statement is why we switched to explicitly passing it in...accountId opens the door to another credential being used, but here, we really need userId.